crypto/x509/v3_purp.c

```diff
@@ -883,6 +888,45 @@ static int check_purpose_timestamp_sign(const X509_PURPOSE *xp, const X509 *x,
     return 1;
 }
 
+/* common code signing checks */
+static int purpose_code_sign(const X509 *x, int require_ca)
+{
+    if (xku_reject(x, XKU_CODE_SIGN))
+        return 0;
+    if (require_ca) {
+        int ca_ret;
+        ca_ret = check_ca(x);
+        if (ca_ret == 0)
+            return 0;
+        /* Check nsCertType if present */
+        if (ca_ret != 5 || x->ex_nscert & NS_OBJSIGN_CA)
```

Comment:
style nit :  to avoid side effects, this should be written: 
`        if (ca_ret != 5 || (x->ex_nscert & NS_OBJSIGN_CA) != 0)`